pkg/lifecycle/api/types.go

```diff
@@ -0,0 +1,51 @@
+package api
+
+import v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```

Comment:
Lets please alias this as `metav1`, otherwise this is really annoying to read as you have to guess what `v1` package the code refers to